SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -559,10 +550,29 @@ theorem msb_sub {x y: BitVec w} :
       = (x.getMsbD 0 ^^ ((~~~y + 1#w).getMsbD 0 ^^ carry (w - 1 - 0) x (~~~y + 1#w) false)) := by
   simp [sub_eq_add_neg, BitVec.neg_eq_not_add, msb_add]
 
-theorem msb_neg {w : Nat} {x : BitVec w} (h : 0 < w) :
-    (~~~x).msb = (getMsbD x 0).not := by
-  rw [BitVec.msb, BitVec.getMsbD_neg]
-  omega
+theorem msb_neg  {x : BitVec w} :
+    (-x).msb = (~~~x + 1#w).msb := by
+  rw [neg_eq_not_add]
+
+theorem getLsbD_neg {i : Nat}  {x : BitVec w} :
+    getLsbD (-x) i = getLsbD (~~~x + 1#w) i := by
+  simp [neg_eq_not_add]
+
+theorem getMsbD_neg {i : Nat} {x : BitVec w} :
+    getMsbD (-x) i = getMsbD (~~~x + 1#w) i := by
+  rw [neg_eq_not_add]
+
+theorem getLsbD_abs {i : Nat} {x : BitVec w} :
+    x.abs.getLsbD i = (if x.msb = true then -x else x).getLsbD i := by
+  simp [BitVec.abs]
+
+theorem getMsbD_abs {i : Nat} {x : BitVec w} :
+    getMsbD (x.abs) i = getMsbD (if x.msb = true then -x else x) i := by
+  simp [BitVec.abs]
+
+theorem msb_abs {w : Nat} {x : BitVec w} :
+    (x.abs).msb = (if x.msb = true then -x else x).msb := by
+  simp [BitVec.abs]
```

Comment:
You might want to be a bit more consistent with wether or not you use dot-notation here